src/bin/pg_autoctl/pgtuning.c

```diff
@@ -262,44 +271,90 @@ pgtuning_edit_guc_settings(GUC *settings, DynamicTuning *tuning,
 
 		if (streq(setting->name, "autovacuum_max_workers"))
 		{
-			appendPQExpBuffer(contents, "%s = %d\n",
-							  setting->name,
-							  tuning->autovacuum_max_workers);
+			if (tuning->autovacuum_max_workers > 0)
+			{
+				appendPQExpBuffer(contents, "%s = %d\n",
+								  setting->name,
+								  tuning->autovacuum_max_workers);
+			}
+			else
+			{
+				appendPQExpBuffer(contents, "%s = %s\n",
+								  setting->name,
+								  setting->value);
+			}
 		}
 		else if (streq(setting->name, "shared_buffers"))
 		{
-			char pretty[BUFSIZE] = { 0 };
-
-			(void) pretty_print_bytes(pretty, sizeof(pretty),
-									  tuning->shared_buffers);
-
-			appendPQExpBuffer(contents, "%s = '%s'\n", setting->name, pretty);
+			if (tuning->shared_buffers > 0)
+			{
+				char pretty[BUFSIZE] = { 0 };
+
+				(void) pretty_print_bytes(pretty, sizeof(pretty),
+										  tuning->shared_buffers);
+
+				appendPQExpBuffer(contents, "%s = '%s'\n",
+								  setting->name, pretty);
+			}
+			else
+			{
+				appendPQExpBuffer(contents, "%s = %s\n",
```

Comment:
Here we now take the settings from our default tuning array:
```
GUC postgres_tuning[] = {
	{ "track_functions", "pl" },
	{ "shared_buffers", "'128 MB'" },
	{ "work_mem", "'4 MB'" },
	{ "maintenance_work_mem", "'64MB'" },
	{ "effective_cache_size", "'4 GB'" },
	{ "autovacuum_max_workers", "3" },
	{ "autovacuum_vacuum_scale_factor", "0.08" },
	{ "autovacuum_analyze_scale_factor", "0.02" },
	{ NULL, NULL }
};

```

We didn't do this before. Do these numbers make sense if we cannot get tuning info? Or would it be better not to tune these settings at all in this case, i.e. removing the `else` case here.

Reply:
Before those changes we would only call the function when we wanted to install locally tuned values. Now we call `pgtuning_prepare_guc_settings` no matter what, and when we see that we're not going to adjust to the host specifics, we use our own default values.

I believe our defaults are either the same as Postgres, or better. So yeah I think it makes sense.